src/prognostic_equations/forcing/external_forcing.jl

```diff
@@ -5,12 +5,75 @@
 import Thermodynamics as TD
 import ClimaCore.Spaces as Spaces
 import ClimaCore.Fields as Fields
+import NCDatasets as NC
+import StatsBase
+import Dierckx
+
+
+"""
+    get_nc_data(ds::NCDatasets.Dataset, var::String)
+
+Returns the data for variable `var`.
+"""
+function get_nc_data(ds, var::String)
+    key_options = (var,)
+    for key in key_options
+        if haskey(ds, key)
+            return ds[key]
+        else
+            for group_option in ["profiles", "reference", "timeseries"]
+                haskey(ds.group, group_option) || continue
+                if haskey(ds.group[group_option], key)
+                    return ds.group[group_option][key]
+                end
+            end
+        end
+    end
+    return nothing
+end
+
+
+function pyinterp(x, xp, fp)
+    spl = Dierckx.Spline1D(xp, fp; k = 1)
+    return spl(vec(x))
+end
+
+mean_nc_data(data, group, var; imin = 100) =
+    StatsBase.mean(data.group[group][var][:, :][:, imin:end], dims = 2)[:]
+init_nc_data(data, group, var) = data.group[group][var][:, :][:, 1]
+
+getvar(var, ds, zc_gcm, zc_les) =
+    pyinterp(zc_gcm, zc_les, mean_nc_data(ds, "profiles", var))
+# nudge to initial LES profiles of u, v, qt, and T (equivalent to GCM initial condition)
+get_nudgevar(var, ds, zc_gcm, zc_les) =
+    pyinterp(zc_gcm, zc_les, init_nc_data(ds, "profiles", var))
+
+"""
+Calculated height dependent relaxation timescale.
+"""
+function compute_gcm_driven_τ(
```

Comment:
Is this function used anywhere? If not, I think it's better to have it in a separate PR that sets nudging timescale.

Reply:
it will be used to compute the τ, but isn't called yet. So I can save it for a separate PR